### PR TITLE
[Community] Support StableDiffusionCanvasPipeline

### PR DESCRIPTION
**Pull request by kadirnar**
I will update the readme after this is merged.

https://github.com/huggingface/diffusers/pull/3586

### CODE REVIEW

**Comment by sayakpaul** [comment]
> @patrickvonplaten , @sayakpaul
> 
> How do I import this [Text2ImageRegion](https://github.com/kadirnar/diffusers/blob/add_canvas_pipeline/examples/community/mixture_canvas.py#L104) class? Should I add it to the pipeline_utils python file?

Sounds like a good idea!

**Comment by patrickvonplaten** [comment]
Hey @kadirnar,

Could you also add a section to the README here: https://github.com/huggingface/diffusers/tree/main/examples/community#community-examples

File: src/diffusers/pipelines/pipeline_utils.py [hunk 1]
```diff
@@ -1537,3 +1539,174 @@ def set_attention_slice(self, slice_size: Optional[int]):
 
         for module in modules:
             module.set_attention_slice(slice_size)
+
+
+@dataclass
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/pipeline_utils.py:1544]: This is way to specific to be in `pipeline_utils.py` can we move this to `mixture_canvas.py` instead? There shouldn't be any changes to `pipeline_utils.py`

File: src/diffusers/pipelines/pipeline_utils.py [hunk 2]
```diff
@@ -31,6 +32,7 @@
 from huggingface_hub import hf_hub_download, model_info, snapshot_download
 from packaging import version
 from requests.exceptions import HTTPError
+from torchvision.transforms.functional import resize
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/pipeline_utils.py:35]: We don't have a dependency on torchvision

File: src/diffusers/pipelines/pipeline_utils.py (hunk 1 above)
Comment (kadirnar) [reply src/diffusers/pipelines/pipeline_utils.py:1544]: Yes I can, but I have to import Image2ImageRegion, Text2ImageRegion, preprocess_image functions. How do I do this?

Example:
```
from diffusers.pipelines.pipeline_utils import Image2ImageRegion, Text2ImageRegion, preprocess_image
```

Comment (patrickvonplaten) [reply src/diffusers/pipelines/pipeline_utils.py:1544]: Could you add all of this to `mixture_canvas.py` ?

**Comment by patrickvonplaten** [comment]
Works for me!

**Comment by kadirnar** [comment]
@patrickvonplaten ,

Thank you, but the [readme.md](https://github.com/huggingface/diffusers/tree/main/examples/community#stable-diffusion-mixture-canvas) file should be updated. Isn't this code wrong?

```
from diffusers.pipelines.pipeline_utils import Image2ImageRegion, Text2ImageRegion, preprocess_image
```

**Comment by patrickvonplaten** [comment]
Wanna open a quick new PR ? :-)

**Comment by patrickvonplaten** [comment]
Ah yeah, I'd add all these "Region" classes directly in the same file so that you don't have to import it

**Pull request merged**